pkg/services/queryhistory/database.go

```diff
@@ -41,6 +43,79 @@ func (s QueryHistoryService) createQuery(ctx context.Context, user *models.Signe
 	return dto, nil
 }
 
+func (s QueryHistoryService) searchQueries(ctx context.Context, user *models.SignedInUser, query SearchInQueryHistoryQuery) ([]QueryHistoryDTO, error) {
+	var queries []QueryHistoryDTO
+
+	if len(query.DatasourceUIDs) == 0 {
+		return nil, errors.New("no selected data source for query history search")
+	}
+
+	if query.Page <= 0 {
+		query.Page = 1
+	}
+
+	if query.Limit <= 0 {
+		query.Limit = 100
+	}
+
+	if query.Sort == "" {
+		query.Sort = "time-desc"
+	}
+
+	err := s.SQLStore.WithDbSession(ctx, func(session *sqlstore.DBSession) error {
+		sql := `SELECT
+			query_history.uid,
+			query_history.datasource_uid,
+			query_history.created_by,
+			query_history.created_at as "created_at",
+			query_history.comment,
+			query_history.queries,
+		`
+
+		if query.OnlyStarred {
+			sql = sql + `1 as "starred"
+				FROM query_history
+				INNER JOIN query_history_star ON query_history_star.query_uid = query_history.uid
+			`
+		} else {
+			sql = sql + `IIF(query_history_star.query_uid IS NULL, false, true) AS starred
```

Comment:
Not sure `IIF` works for sqlite, postgres and mysql.

Reply:
Think at least postgres might need the `CASE` clause. With my suggested changes should run your tests using mysql and postgres tests in CI

Reply:
Thank you so much suggesting `CASE`. You are right, Postgres needs `CASE` instead of `IIF`.

Reply:
Updated in https://github.com/grafana/grafana/pull/45932/commits/e3fbbf33023229eae22acf1c90edda70300b2eb9